home/management/commands/updatecourses.py

```diff
@@ -0,0 +1,104 @@
+            s = Semester(semester)
+            kwargs = {"semester": semester, "per_page": 100, "page": 1}
+            course_data = requests.get("https://api.umd.io/v1/courses", params=kwargs).json()
+
+            if "error_code" in course_data[0].keys():
+                print(f"umd.io doesn't have data for {s.name()}!")
+                continue
+
+            print(f"Working on courses for {s.name()}...")
+
+            while course_data:
+                for umdio_course in course_data:
+                    course = Course.unfiltered.filter(name=umdio_course['course_id']).first()
+                    if not course:
+                        course = Course(
+                            name=umdio_course['course_id'],
+                            department=umdio_course['dept_id'],
+                            course_number=umdio_course['course_id'][4:],
+                            title=umdio_course['name'],
+                            credits=umdio_course['credits'],
+                            description=umdio_course["description"]
+                        )
+
+                        course.save()
+                        self.total_num_new_courses += 1
+
+                    self._professors(course, s)
+                    print(course)
+
+                kwargs["page"] += 1
+                course_data = requests.get("https://api.umd.io/v1/courses", params=kwargs).json()
+
+        print(f"\n** New Courses Created: {self.total_num_new_courses} **")
+        print(f"** New Professors Created: {self.total_num_new_professors} **")
+
+        runtime = datetime.now() - t_start
+        print(f"Runtime: {round(runtime.seconds / 60, 2)} minutes")
+
+    def _professors(self, course: Course, semester: Semester):
+        kwargs = {"course_id": course.name}
+        umdio_professors = requests.get("https://api.umd.io/v1/professors", params=kwargs).json()
+
+        for umdio_professor in umdio_professors:
+            try:
+                professor = Professor.unfiltered.filter(name=umdio_professor['name']).first()
+            except TypeError:
+                continue
+
+            if umdio_professor['name'] == "Instructor: TBA":
+                continue
+
+            if not professor:
+                # To make our lives easier, attempt to automatically verify the professor
+                # following the same criteria in admin.py
+                split_name = umdio_professor['name'].strip().split()
+                first_name = split_name[0].lower().strip()
+                last_name = split_name[-1].lower().strip()
+                query = Professor.verified.filter(
+                    (
+                        Q(name__istartswith=first_name) &
+                        Q(name__iendswith=last_name)
+                    ) |
+                    Q(slug="_".join(reversed(split_name)).lower())
+                )
+
+                prof_name = " ".join([name.capitalize() for name in umdio_professor['name'].split()])
```

Comment:
why are you capitalizing the name from umdio? umdio pulls from soc, so shouldn't that be the single source of truth for the capitalization of a professor's name?

Reply:
Some names from soc are in all caps for some reason so I do that to normalize the names in our db

Reply:
I would've but I've already run the script locally (took about an hour for 5 semesters) so all the names have already been normalized

Reply:
Thought about this some more and this is going to cause issues with professors who have capitals in the middle of their name. Many french names have multiple capitals for instance (including my own, `DeVoe`).

I'd need to see an example of an all-capitalized name on testudo (which seems very weird to me) to comment on what we should be doing instead.

Reply:
Similarly, for professors with hyphens where both constituent parts are capitalized: https://planetterp.com/professor/naharro-calderon

Reply:
valid point. I suppose names appearing in all caps doesn't occur very often anyway so we can manually change the name if need be